src/snapshot_middleware/project.rs

```diff
@@ -78,7 +80,42 @@ pub fn snapshot_project_node(
         // If the path specified in the project is relative, we assume it's
         // relative to the folder that the project is in, project_folder.
         let path = if path.is_relative() {
-            Cow::Owned(project_folder.join(path))
+            // Convert paths to use forward slashes for compatibility with the relative-path crate
+            let project_folder_with_slash_separator = match project_folder.to_slash() {
```

Comment:
Agreed - the relative-path crate is painful with regards to converting back and forth. It's a shame that rust doesn't have a built-in function for normalising paths.

Neither of these libraries (as used here) will handle symlinks as they are not hitting the FS to do any checks. Thinking about contrived symlink cases:

1. Source project is loaded from under a symlink.
```
/foo/default.project.json: $path: ../world
/bar/baz: symlink to /foo
load(/bar/baz/default.project.json): /bar/baz/../world -> /bar/world
load(/foo/default.project.json): /foo/../world -> /world
```
2. Intermediate directory in $path is a symlink.
```
/foo/default.project.json: $path: ../bar/../world
/bar: symlink to /biff/baz
/biff/baz
load(/foo/default.project.json): /foo/../bar/../world ->Unclear whether this should resolve to /world or /biff/world

Perhaps we should go with the components stack approach, but when processing the components check if any component is a symlink and abort the normalisation if it is, instead returning the original path. This shouldn't compromise achieving the goal of the solution because symlinks are less common on Windows anyway, which is the only platform where we need this.

Reply:
That sounds like a reasonable approach! If we adjust it so we only check for symlinks _specifically on the components we want to normalize away_, then even if you are within a symlinked directory, it doesn't matter if you don't need to climb that far

Reply:
If we go down the road of checking for symlinks, we'll likely need to expand the API surface of `memofs`. We can't use the stock `fs` APIs because a project could be built from an in-memory virtual filesystem, like with `rojo plugin install`.